tendrl/node_agent/manager/manager.py

```diff
@@ -227,6 +228,34 @@ def on_pull(self, raw_data):
                     node_id=raw_data["node_id"],
                 )
             )
+        if "disks" in raw_data:
+            try:
+                self.etcd_client.delete(
+                    ("nodes/%s/Disks") % raw_data["node_id"], recursive=True)
+            except Exception as ex:
```

Comment:
When node agent start first time, there is no disk details in etcd. Code for push disk details is under this code. If it try to delete disk details from etcd, there no disk details pushed in etcd yet, so it raise exception and Because of that exception insert code never execute and control came out from function. This is the reason i catch exception here. Next time etcd details are present in etcd so it will delete old details and new disk details are pushed.